examples/mem-event-example.c

```diff
@@ -39,10 +37,26 @@ static void close_handler(int sig)
     interrupted = true;
 }
 
-event_response_t mem_cb(vmi_instance_t vmi, vmi_event_t *event)
+event_response_t singlestep_cb(vmi_instance_t vmi, vmi_event_t *event)
 {
-    (void)vmi;
+    addr_t paddr = 0;
+    // We need to know the physical address of the event that triggered this callback to be able to
+    // reinstate the original memory permissions
+    if (vmi_pagetable_lookup(vmi, event->x86_regs->cr3, event->ss_event.gla, &paddr) == VMI_FAILURE) {
```

Comment:
IMHO it makes a lot more sense to just stuff this information in `event->data` using GSIZE_TO_POINTER(event->mem_event.gfn) when the permissions are getting cleared. Saves you a pagetable walk.

Reply:
I guess it makes sense to get rid of the pagetable walk. I was trying to avoid having to put the events into global scope.
`GSIZE_TO_POINTER` expands to a simple type cast to `gpointer`, right? What's the benefit over using `(void *)`? Doesn't it only make sense if we are working with glibc objects?

Reply:
It's more portable, see https://developer.gimp.org/api/2.0/glib/glib-Type-Conversion-Macros.html#GSIZE-TO-POINTER:CAPS